internal/xds/translator/authentication.go

```diff
@@ -48,10 +48,6 @@ func patchHCMWithJwtAuthnFilter(mgr *hcm.HttpConnectionManager, irListener *ir.H
 		return errors.New("ir listener is nil")
 	}
 
-	if len(irListener.Routes) == 0 {
-		return errors.New("ir listener contains no routes")
```

Comment:
Instead of returning an error, the func should just return early since other parts of the code path are expecting the listener to contain routes. Otherwise, please update the other authn funcs to check for `irListener.Routes` and return early.

Reply:
looks like its being handled already https://github.com/envoyproxy/gateway/blob/38a6dddd21efd0edb1deade116d771f8252e1ad7/internal/xds/translator/authentication.go#L515